pyproject.toml

```diff
@@ -117,3 +117,64 @@ Source = "https://github.com/caikit/caikit"
 markers = [
     "examples: marks tests as e2e examples (deselect with '-m \"not examples\"')",
 ]
+
+
+[tool.ruff]
+line-length = 100
+target-version = "py38"
+exclude = ["caikit/runtime/protobufs/*.py"]
+
+
+[tool.ruff.lint]
+select = [ "E", "F", "UP", "B", "SIM", "I"]
```

Comment:
`select` marks the rules that we do want to check the code for. Here's an extract from the ruff docs

```toml
[tool.ruff.lint]
select = [
  # pycodestyle
  "E",
  # Pyflakes
  "F",
  # pyupgrade
  "UP",
  # flake8-bugbear
  "B",
  # flake8-simplify
  "SIM",
  # isort
  "I",
]

[List of available rules](https://docs.astral.sh/ruff/rules/)